app/src/main/java/com/spencerpages/MainActivity.java

```diff
@@ -984,33 +959,145 @@ private void handleExportCollectionsPart2(){
         				break;
         			}
 
-        		}while(resultCursor.moveToNext());
+        		} while(resultCursor.moveToNext());
         	}
         	resultCursor.close();
-        	
-        	try {
-    			out.close();
-    		} catch (IOException e) {
-    			showCancelableAlert(mRes.getString(R.string.error_closing_output_file, file.getAbsolutePath()));
-    			return;
-    		}
-        	
+
+        	if(closeOutputFile(out, outputFile)){
+        	    return;
+        	}
+
         	if(errorOccurred){
         		return;
         	}
     	}
-    	
+
     	mDbAdapter.close();
-    	
+
 		showCancelableAlert(mRes.getString(R.string.success_export, EXPORT_FOLDER_NAME));
     }
-    
+
     private void showCancelableAlert(String text) {
 		mBuilder.setMessage(text).setCancelable(true);
 		AlertDialog alert = mBuilder.create();
 		alert.show();
 	}
 
+    private BufferedReader openFileForReading(File file){
+
+        FileInputStream fileInputStream;
+        InputStreamReader inputStreamReader;
+        BufferedReader bufferedReader;
+
+        try {
+            fileInputStream = new FileInputStream(file);
+            inputStreamReader = new InputStreamReader(fileInputStream, "UTF8");
+            bufferedReader = new BufferedReader(inputStreamReader);
+        } catch (Exception e) {
+            if(BuildConfig.DEBUG) {
+                Log.e(MainApplication.APP_NAME, e.toString());
+            }
+            showCancelableAlert(mRes.getString(R.string.error_open_file_reading, file.getAbsolutePath()));
+            return null;
+        }
+        return bufferedReader;
+    }
+
+    private boolean closeInputFile(BufferedReader in, File file){
+        return closeInputFile(in, file, false);
+    }
+
+    private boolean closeInputFile(BufferedReader in, File file, boolean silent){
+        try {
+            in.close();
+        } catch (IOException e) {
+            if(!silent){
+                showCancelableAlert(mRes.getString(R.string.error_closing_input_file, file.getAbsolutePath()));
+            }
+            return true;
+        }
+        return false;
+    }
+
+    private OutputStreamWriter openFileForWriting(File file){
+
+        FileOutputStream fileOutputStream;
+        OutputStreamWriter outputStreamWriter;
```

Comment:
These can be declared in the try block (and outputStreamWriter returned from there)